tools/bin/ariaAOIassist.py

```diff
@@ -488,6 +503,67 @@ def __polygonFromFrame__(self,frame):
                     (float(frame['Near End Lon']),float(frame['Near End Lat']))])
         return P
 
+
+    ## Suggested AOI based on intersection of all polygons
+    def intersectionAOI(self):
+        '''
+            Create a "suggested" AOI based on the intersection of all frame polygons.
+            This function does not guarantee a result.
+        '''
+        # Exclude dates based on spatial criteria if not already done
+        if self.partialDatesRemoved==False:
+            self.checkContinuity(removeIncompleteDates=True)
+
+        # Check that some valid dates remain
+        errMessage='''
+No valid dates were found to meet all spatial and temporal criteria.
+Suggest to narrow the latitude bounds using the --lat_bounds \'min max\' option, or
+exclude dates using the --exclude_dates option.
+Check the {}_lat_extents.eps file for a visual aid using the --flag_partial_coverage option.
+        '''.format(self.trackCode)
+        assert self.nRemainingDates>0, errMessage
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=5801844)